library/std/src/fs.rs

```diff
@@ -2980,6 +2980,19 @@ pub fn read_dir<P: AsRef<Path>>(path: P) -> io::Result<ReadDir> {
 ///
 /// [changes]: io#platform-specific-behavior
 ///
+/// # Symlinks
+/// On UNIX systems, it is impossible to manipulate the permission bits of a symlink itself[^1].
```

Comment:
The language is too strong, https://superuser.com/a/1188707

And it's leading with the wrong thing, we should first say that it follows symlinks and what implications that has. The why would be more of a footnote.